src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayMiscellaneousOptions.cs

```diff
@@ -72,5 +72,11 @@ public enum SymbolDisplayMiscellaneousOptions
         /// Append '!' to non-nullable reference types.
         /// </summary>
         IncludeNotNullableReferenceTypeModifier = 1 << 8,
+
+        /// <summary>
+        /// Insert a tuple into the display parts as a single part instead of multiple parts (similar
+        /// to how anonymous types are inserted).
+        /// </summary>
+        CollapseTupleTypes = 1 << 9,
```

Comment:
>CollapseTupleTypes [](http://example.com/codeflow?start=8&length=18)

I'm curious why special-case tuples and anonymous types. Wouldn't similar behavior be useful for other type arguments?

Reply:
Primarily because they are structural types that require a lot more than a name to reference. For example, when you have a nominal type, it doesn't matter how many members that type has, you just reference it by name. Hit with a structural type you have to include that structure in somehow. 

If we have more cases like that in the future, I'd expect we go down a similar route.

Reply:
> when you have a nominal type, it doesn't matter how many members that type has, you just reference it by name

But the nominal type could be arbitrarily complicated, for instance `Dictionary<string, object>`.

Reply:
Potentially. We haven't had that arise as significant enough to need attention. If we did though, I'd prefer that as a separate flag.